src/poetry/repositories/pool.py

```diff
@@ -135,18 +135,15 @@ def package(
             raise ValueError(f'Repository "{repository}" does not exist.')
 
         if repository is not None and not self._ignore_repository_names:
-            package = self.repository(repository).package(name, version, extras=extras)
-            if package is not None:
-                return package
-        else:
-            for repo in self._repositories:
-                try:
-                    package = repo.package(name, version, extras=extras)
-                except PackageNotFound:
-                    continue
-
-                if package:
-                    return package
+            return self.repository(repository).package(name, version, extras=extras)
+
+        for repo in self._repositories:
+            try:
+                package = repo.package(name, version, extras=extras)
+            except PackageNotFound:
+                continue
```

Comment:
Why this change in logic removing the else?

Reply:
this doesn't change the logic: if we go into the if branch, then we either return or we raise an exception.  The 'else' is now redundant.